### PR TITLE
Emit Clone methods in abstract records as abstract

### LINKED ISSUE #44890: Abstract record class results in invalid code
Compile:
```C#
abstract record C;
```
peverify.exe:
```
[IL]: Error: [C::Clone][offset 0x00000001] Cannot construct an instance of abstract class.
```

### PR DESCRIPTION
**Pull request by agocke**
The clone method is designed to call the containing type's
constructor, but this can't be done when the containing type
is abstract. A reasonable alternative is to make the method
abstract.

Fixes #44890

### CODE REVIEW

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs [hunk 1]
```diff
@@ -1315,5 +1315,73 @@ public void RecordCloneSymbol()
...
+                parseOptions: TestOptions.RegularPreview,
+                options: TestOptions.ReleaseExe);
+
+            var r = comp.GlobalNamespace.GetTypeMember("R");
+            var clone = (MethodSymbol)r.GetMembers(WellKnownMemberNames.CloneMethodName).Single();
+            Assert.False(clone.IsOverride);
+            Assert.False(clone.IsVirtual);
+            Assert.True(clone.IsAbstract);
+            Assert.Equal(0, clone.ParameterCount);
+            Assert.Equal(0, clone.Arity);
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1347]: Consider checking display string: `Assert.Equal("... <>Clone(...)", clone.ToTestDisplayString());` in each case.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs [hunk 2]
```diff
@@ -1315,5 +1315,73 @@ public void RecordCloneSymbol()
...
+            verifier.VerifyIL("C.Main", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  newobj     ""R3..ctor()""
+  IL_0005:  callvirt   ""R R.<>Clone()""
+  IL_000a:  pop
+  IL_000b:  ret
+}");
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1385]: Consider testing concrete base, abstract derived type:
```C#
record R;
abstract R2 : R;
record R3 : R2;
```

**Comment by gafter** [comment]
These changes are not supported by the specification.

**Comment by agocke** [comment]
@gafter There are a lot changes we need to put into the spec, but I've pulled out this one since it's pretty straightforward https://github.com/dotnet/csharplang/pull/3547

**Comment by gafter** [comment]
> @gafter There are a lot changes we need to put into the spec

Agreed, and yet it hasn't been done.

I would expect that to be easier than implementing the changes. It is necessary to review the changes. LDM notes in isolation aren't enough to review compiler changes.

**Comment by agocke** [comment]
@gafter Yup, the preview is snapping today so we should have more time to update the spec now. This change seems fully described by the proposed spec change, though.

**Pull request merged**